packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/AppConfigurationImpl.kt

```diff
@@ -158,11 +159,12 @@ public class AppConfigurationImpl @OptIn(ExperimentalKBsonSerializerApi::class)
     }
 
     private fun initializeSyncClientConfig(
+        appConfigPointer: RealmAppConfigurationPointer,
         webSocketTransport: WebSocketTransport?,
         sdkInfo: String?,
         applicationInfo: String?
     ): RealmSyncClientConfigurationPointer =
-        RealmInterop.realm_sync_client_config_new()
+        RealmInterop.realm_app_config_get_sync_client_config(appConfigPointer)
```

Comment:
This function appears to be unimplemented and I'm not even sure how it could be implemented considering that `AppConfig::SyncClientConfig` is a simple struct and not a pointer, so getting it would just give us a copy. I've opened https://github.com/realm/realm-core/pull/7792 to discuss how this is intended to be used by consumers.

Reply:
https://github.com/realm/realm-core/pull/7792 has been superseded by https://github.com/realm/realm-core/pull/7891 that adds the implementation for `realm_app_config_get_sync_client_config` and is now merged.